src/script/index.js

```diff
@@ -83,9 +83,13 @@ function setMolecule(molString) {
 	$.ajax({
 		type: "POST",
 		url: '/plexus/rest-v0/util/calculate/stringMolExport',
-		data: { structure: molString, parameters: 'MOL' }
-	}).done(function updateInputCompound(smiles) {
-		ketcher.ui.load(smiles, {
+		data: { structure: molString, parameters: 'MOL:V3' }
+	}).done(function updateInputCompound(molfile) {
+		ketcher.ui.load(molfile, {
+			rescale: true
+		});
+	}).fail(function onError() {
+		ketcher.ui.load(molString, {
```

Comment:
un-related change:
1. tries to import the supplied representation by first converting it to MOL V3000 using the plexus convert service
2. if not possible, it assumes the supplied representation to already be a mol representation, and directly imports it.

Reply:
how is error handling managed if the service fails and the representation is not mol?

Reply:
It has an error handling mechanism down in the flow, inside [structformat#fromString](https://github.com/schrodinger/ketcher/blob/master/src/script/ui/data/convert/structformat.js#L123)